### PR TITLE
fix(MX-381): validate transfer source ownership

### PR DESCRIPTION
**Pull request by AnvayKharb**
## Summary by CodeRabbit

* **Bug Fixes**
  * Prevented potential null errors during transfer-type validation.
  * Updated transfer quote/prepare/confirm flows to use the validated source client for fee calculation, OTP handling, and notifications.
  * Added stronger source ownership validation and standardized validation failures to block foreign/invalid sources.

* **Tests**
  * Added comprehensive JUnit/Mockito coverage for prepare, quote, confirm, and OTP resend flows, including edge cases and “no side effects” interaction assertions.

### CODE REVIEW

File: src/main/java/org/apache/fineract/selfservice/account/service/SelfAccountTransferWritePlatformServiceImpl.java [hunk 1]
```diff
@@ -2070,32 +2076,88 @@ private void validateSufficientFunds(
...
 
-    String result = useOriginal
+    String result =
+        useOriginal
             ? description
             : (description != null && !description.isBlank() ? description : "Transfer")
-              + " via "
-              + (mode != null ? mode : "");
-    
-    return String.format("%-15s", result);   // right-pad with spaces
```
Comment (IOhacker) [inline src/main/java/org/apache/fineract/selfservice/account/service/SelfAccountTransferWritePlatformServiceImpl.java:2099]: We require to have at least 15 characters in the transfer description, it is required by fast payments.

**Review by IOhacker — APPROVED** [review]
LGTM
